ntoskrnl/ps/security.c

```diff
@@ -670,22 +672,61 @@ PsImpersonateClient(IN PETHREAD Thread,
             }
         }
 
-        /* FIXME: If the process token can't impersonate, we need to make a copy instead */
+        /*
+         * Assign the token we get from the caller first. The reason
+         * we have to do that is because we're unsure if we can impersonate
+         * in the first place. In the scenario where we cannot then the
+         * last resort is to make a copy of the token and assign that newly
+         * token to the impersonation information.
+         */
+        TokenImpersonation = Token;
```

Comment:
Would `ImpersonationToken` be a better name? `TokenImpersonation` appears to be an enum member.